### PR TITLE
632 hello world

### PR DESCRIPTION
**Pull request by Silver92**
A new job dealing with multiple inputs and outputs. The job supports different job types and a maximum of 10 inputs and 10 outputs for each job type. But it does not support the job list type.

### CODE REVIEW

File: rodan/jobs/helloworld/helloworld.py [hunk 1]
```diff
@@ -10,10 +10,20 @@ class HelloWorld(RodanTask):
...
+        },
     )
     output_port_types = (
-        {'name': 'Text output', 'minimum': 1, 'maximum': 1, 'resource_types': ['text/plain']},
+        {
+            'name': 'Text output', 
+            'minimum': 1, 
+            'maximum': 1, 
+            'resource_types': ['text/plain']
+        },
```
Comment (lingxiaoyang) [inline rodan/jobs/helloworld/helloworld.py]: 🚫 Please remove these reformatting changes from this PR. These changes are unrelated and have a higher chance of conflicts when we merge with DDMAL later on. 

I know they are causing linter errors. I'll discuss with DDMAL at a later time on a dedicated effort to reformat the code.

File: rodan/jobs/helloworld/helloworld.py [hunk 2]
```diff
@@ -29,6 +39,46 @@ def run_my_task(self, inputs, settings, outputs):
...
+    )
+
+    def run_my_task(self, inputs, settings, outputs):
+        for output_type in outputs:
+            for output in outputs[output_type]:
+                outfile = open(output["resource_path"], "w")
+                # Write input file infomation into the current 
+                # out file.
```
Comment (lingxiaoyang) [inline rodan/jobs/helloworld/helloworld.py]: ℹ️ I guess you were getting a "line too long" error if you put them on a single line. The PEP 8 style guide recommends 80 characters per line, but it is too restrictive in practice and we commonly use 120 characters.

File: rodan/jobs/helloworld/helloworld.py [hunk 3]
```diff
@@ -29,6 +39,46 @@ def run_my_task(self, inputs, settings, outputs):
...
+            'minimum': 1, 
+            'maximum': 10, 
+            'resource_types': ['text/plain']
+        },
+    )
+
+    def run_my_task(self, inputs, settings, outputs):
+        for output_type in outputs:
+            for output in outputs[output_type]:
+                outfile = open(output["resource_path"], "w")
```
Comment (lingxiaoyang) [inline rodan/jobs/helloworld/helloworld.py:61]: ℹ️ You can leverage context managers:

```python
with open(output["resource_path"], "w") as outfile:
    outfile.write(...)
```

File: rodan/jobs/helloworld/helloworld.py [hunk 4]
```diff
@@ -29,6 +39,46 @@ def run_my_task(self, inputs, settings, outputs):
...
+                # Write input file infomation into the current 
+                # out file.
+                for input_type in inputs:
+                    for input in inputs[input_type]:
+                        infile = open(input["resource_path"], "r")
+                        outfile.write(infile.read() + "\n")
+                        infile.close()
```
Comment (lingxiaoyang) [inline rodan/jobs/helloworld/helloworld.py]: ⚠️ Heads-up: this loop is unrelated to output and can surely be moved outside.

**Review by lingxiaoyang — CHANGES_REQUESTED** [review]
The new class looks good to me. Please remove the unrelated changes.

Comments on your commit message:
1. It's too long. Please keep it within 72 characters on the first line. You can put all other things on paragraphs below, with a blank line in between the first line.
2. Use the imperative mood, like "Add a test job with multi input and output".
image

Then we're good to merge.

File: rodan/jobs/helloworld/helloworld.py (hunk 3 above)
Comment (Silver92) [reply rodan/jobs/helloworld/helloworld.py:61]: Yes, I used it the first time, but it looks like too many indentions and complicated to read. That's why I try to use the traditional way.

File: rodan/jobs/helloworld/helloworld.py (hunk 4 above)
Comment (Silver92) [reply rodan/jobs/helloworld/helloworld.py]: I can extract an additional method of reading the input file

File: rodan/jobs/helloworld/helloworld.py [hunk 5]
```diff
@@ -29,6 +29,47 @@ def run_my_task(self, inputs, settings, outputs):
...
+            'maximum': 10, 
+            'resource_types': ['text/plain']
+        },
+    )
+
+    def run_my_task(self, inputs, settings, outputs):
+        for output_type in outputs:
+            for output in outputs[output_type]:
+                outfile = open(output["resource_path"], "w")
+                self.WriteInputFile(inputs, outfile)
```
Comment (lingxiaoyang) [inline rodan/jobs/helloworld/helloworld.py]: 🚫 Lower case `w`.

🚫  In Python, we usually use snake_case as variable/function names and use CamelCase for classes. Please rename it as `write_input_file`.

**Review by lingxiaoyang — CHANGES_REQUESTED** [review]
🚫 There are 8 commits, containing merge commits that shouldn't appear in a PR.

I see reverts in the history, but revert is not rebase. Can you **squash** all of them into a single commit?

**Review by lingxiaoyang — CHANGES_REQUESTED** [review]
🚫 The 9 commits should be squashed into a single one.

**Comment by lingxiaoyang** [comment]
Replaced by #6.
